Automattic/jetpack
# Remove pnpm hoisting for eslint

#### Changes proposed in this Pull Request:

This involves moving the bits of eslint configuration that load plugins
or sharable configs into `tools/js-tools/eslintrc/`, so the plugins and
configs can be loaded relative to that directory instead of relative to
wherever the origin `.eslintrc.js` file happens to be.

This also drops the dep on `@sveltejs/eslint-config`, which hasn't been
used since #23482.

And for some reason projects/js-packages/shared-extension-utils/ had
both .eslintrc.js and .eslintrc.cjs, with identical content. So I
deleted one.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Do `pnpm install`, then check that `node_modules/.pnpm/node_modules/` no longer exists. 
* `pnpm lint` should complete and give the same list of warnings and errors that it does in master.
* For any directory with an eslintrc file, `pnpm lint-file path/to/dir` should complete and give the same list of warnings and errors that it does in master. Except for `projects/plugins/jetpack/modules/widget-visibility/editor`, which is currently broken in master and is fixed here.
* Someone with vscode should check that vscode's eslinting still works too.
